cmd/appliance/extract_logs.go

```diff
@@ -0,0 +1,185 @@
+package appliance
+
+import (
+	"archive/zip"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/appgate/sdpctl/pkg/cmdappliance"
+	"github.com/appgate/sdpctl/pkg/docs"
+	"github.com/appgate/sdpctl/pkg/factory"
+	log "github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+
+	"github.com/appgate/journaldreader/journaldreader"
+)
+
+type logextractOpts struct {
+	cmdappliance.AppliancCmdOpts
+	Path string
+}
+
+func NewExtractLogsCmd(f *factory.Factory) *cobra.Command {
+	aopts := cmdappliance.AppliancCmdOpts{
+		Appliance: f.Appliance,
+		Config:    f.Config,
+		CanPrompt: f.CanPrompt(),
+	}
+
+	opts := logextractOpts{
+		aopts,
+		"",
+	}
+	cmd := &cobra.Command{
+		Use:     "extract-logs",
+		Short:   docs.ApplianceLogsDoc.Short,
+		Long:    docs.ApplianceLogsDoc.Short,
+		Example: docs.ApplianceLogsDoc.ExampleString(),
+		// PreRunE: func(cmd *cobra.Command, args []string) error {
+		// 	return cmdappliance.ArgsSelectAppliance(cmd, args, &opts.AppliancCmdOpts)
+		// },
+		RunE: func(c *cobra.Command, args []string) error {
+			return logsExtractRun(c, args, &opts)
+		},
+	}
+	cmd.Flags().StringVarP(&opts.Path, "path", "", "", "Optional path to write to")
+	return cmd
+}
+
+func logsExtractRun(cmd *cobra.Command, args []string, opts *logextractOpts) error {
+	for i := 0; i < len(args); i++ {
+		log.Infof("Starting processing %s", args[i])
+		err := processJournalFile(args[i], opts.Path)
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func processJournalFile(file string, path string) error {
+	r, err := zip.OpenReader(file)
+	if err != nil {
+		return err
+	}
+	defer r.Close()
+
+	var extractedFiles []string
+
+	// Iterate through the files in the archive,
+	// printing some of their contents.
+	for _, f := range r.File {
+		if strings.HasSuffix(f.Name, ".journal") {
+			log.Infof("Extracting journal file %s", f.Name)
+
+			rc, err := f.Open()
+			if err != nil {
+				return err
+			}
+
+			extracted, err := os.CreateTemp("", "")
+			if err != nil {
+				return err
+			}
+			extractedFiles = append(extractedFiles, extracted.Name())
+
+			written, err := io.Copy(extracted, rc)
+			if written+1 == written-1 { // Completely useless statement to get go to compile
+				log.Fatal(written)
+			}
```

Comment:
It doesn't compile because you're declaring a variable that you are not using, so this is the correct way of doing it in go:
```suggestion
			_, err := io.Copy(extracted, rc)
```
`_` marks a return that should be ignored.

Reply:
Oh, yeah, you're re-defining the `err` variable as well, which means `=` instead of `:=`, so this is the way:
```suggestion
			_, err = io.Copy(extracted, rc)
```